tests/phpunit/api/v3/ContributionTest.php

```diff
@@ -1748,6 +1748,46 @@ public function testCompleteTransaction() {
   }
 
   /**
+   * Test completing a transaction via the API with a non-USD transaction.
+   */
+  public function testCompleteTransactionEuro() {
+    $mut = new CiviMailUtils($this, TRUE);
+    $this->swapMessageTemplateForTestTemplate();
+    $this->createLoggedInUser();
+    $params = array_merge($this->_params, array('contribution_status_id' => 2, 'currency' => 'EUR'));
+    $contribution = $this->callAPISuccess('contribution', 'create', $params);
+
+    $this->callAPISuccess('contribution', 'completetransaction', array(
+      'id' => $contribution['id'],
+    ));
+
+    $contribution = $this->callAPISuccess('contribution', 'getsingle', array('id' => $contribution['id']));
+    $this->assertEquals('SSF', $contribution['contribution_source']);
+    $this->assertEquals('Completed', $contribution['contribution_status']);
+    $this->assertEquals(date('Y-m-d'), date('Y-m-d', strtotime($contribution['receipt_date'])));
+
+    $entityFinancialTransactions = $this->getFinancialTransactionsForContribution($contribution['id']);
+    $entityFinancialTransaction = reset($entityFinancialTransactions);
+    $financialTrxn = $this->callAPISuccessGetSingle('FinancialTrxn', array('id' => $entityFinancialTransaction['financial_trxn_id']));
+    $this->assertEquals('EUR', $financialTrxn['currency']);
+
+    $mut->checkMailLog(array(
+      'email:::anthony_anderson@civicrm.org',
+      'is_monetary:::1',
+      'amount:::100.00',
```

Comment:
Yeah maybe - that part of the test is pretty tangental to what I was trying to test but since similar tests had it I kept it there. Once we drop php 5.3 support I think a good cleanup of the test classes would be in order (with traits)